patches/server/0723-Fix-commands-from-signs-not-firing-command-events.patch

```diff
@@ -0,0 +1,103 @@
++        this.delegate = delegate;
++    }
++
++    @Override
++    public void sendMessage(Component message, UUID sender) {
++        delegate.sendMessage(message, sender);
++    }
++
++    @Override
++    public boolean acceptsSuccess() {
++        return delegate.acceptsSuccess();
++    }
++
++    @Override
++    public boolean acceptsFailure() {
++        return delegate.acceptsFailure();
++    }
++
++    @Override
++    public boolean shouldInformAdmins() {
++        return delegate.shouldInformAdmins();
++    }
++
++    @Override
++    public CommandSender getBukkitSender(CommandSourceStack wrapper) {
++        return delegate.getBukkitSender(wrapper);
++    }
++}
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java
+index 9b5d11ece006d7aa893360a84ba652c666517ac1..81739da3afd5979ecf745dce39d547c8dcc07250 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java
+@@ -227,7 +227,15 @@ public class SignBlockEntity extends BlockEntity implements CommandSource { // C
+             ClickEvent chatclickable = chatmodifier.getClickEvent();
+ 
+             if (chatclickable != null && chatclickable.getAction() == ClickEvent.Action.RUN_COMMAND) {
+-                player.getServer().getCommands().performCommand(this.createCommandSourceStack(player), chatclickable.getValue());
++                // Paper start
++                net.minecraft.server.network.ServerGamePacketListenerImpl.LOGGER.info(player.getScoreboardName() + " issued server command: /" + chatclickable.getValue());
++                org.bukkit.event.player.PlayerCommandPreprocessEvent event = new org.bukkit.event.player.PlayerCommandPreprocessEvent(player.getBukkitEntity(), "/" + chatclickable.getValue(), new org.bukkit.craftbukkit.util.LazyPlayerSet(player.getServer()));
++                final String command = event.getMessage().substring(1);
++                if (!event.callEvent()) {
++                    return false;
++                }
++                player.getServer().getCommands().performCommand(this.createCommandSourceStack(((org.bukkit.craftbukkit.entity.CraftPlayer) event.getPlayer()).getHandle()), command, command, false);
++                // Paper end
+             }
+         }
+ 
+@@ -263,8 +271,21 @@ public class SignBlockEntity extends BlockEntity implements CommandSource { // C
+         String s = player == null ? "Sign" : player.getName().getString();
+         Object object = player == null ? new TextComponent("Sign") : player.getDisplayName();
+ 
++        // Paper start - send messages back to the player
++        CommandSource commandSource = new io.papermc.paper.commands.DelegatingCommandSource(this) {
++            @Override
++            public void sendMessage(net.minecraft.network.chat.Component message, java.util.UUID sender) {
```

Comment:
Why are these overrides needed here if they are overridden in DelegatingCommandSource

Reply:
Because it's redirecting the message to be send to the player instead of the delegate CommandSource. In this case, the SignBlockEntity.